include/glow/Base/Type.h

```diff
@@ -371,6 +378,9 @@ struct Type final {
     switch (Ty) {
     case ElemKind::FloatTy:
       return sizeof(float);
+    case ElemKind::Float16Ty:
+      static_assert(sizeof(float16_t) == 2, "Half precision should be 16-bit");
```

Comment:
would it be better to move this assert close to the aliasing on top?

Reply:
Works for me! Done.